src/main/java/org/tikv/common/TiSession.java

```diff
@@ -59,7 +59,9 @@
   private volatile Catalog catalog;
   private volatile ExecutorService indexScanThreadPool;
   private volatile ExecutorService tableScanThreadPool;
+  private volatile ExecutorService batchGetThreadPool;
```

Comment:
https://github.com/tikv/client-java/blob/master/src/main/java/org/tikv/common/KVClient.java#L137 should use this thread pool

Reply:
fixed